src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -396,10 +303,7 @@ private void Register(Type t, DeepCopier cop, Serializer ser, Deserializer deser
                 {
                     registeredTypes.Add(t);
                     string name = t.OrleansTypeKeyString();
-                    lock (types)
-                    {
-                        types[name] = t;
-                    }
+                    types[name] = t;
                     if (cop != null)
                     {
                         lock (copiers)
```

Comment:
are all of these locks still required? Maybe it makes sense to use the `CachedReadConcurrentDictionary` if you expect some minor concurrency during warmup?

Reply:
BTW, fine if we postpone all the locks removals from this class into a separate PR that we can test in isolation. It's probably a lot more appropriate to do it that way

---
In reply to: [143850025](https://github.com/dotnet/orleans/pull/3518#discussion_r143850025) [](ancestors = 143850025)

Reply:
Let's postpone it until afterwards